moshi/src/main/java/com/squareup/moshi/BufferedSourceJsonReader.java

```diff
@@ -236,6 +237,11 @@
   }
 
   private int doPeek() throws IOException {
+    if (stackSize == 1
+            && buffer.request(BYTE_ORDER_MARK.size())
```

Comment:
This should be `source.request`. We need to fill `buffer` by requesting from `source`